store/tikv/snapshot.go

```diff
@@ -285,10 +298,17 @@ func (s *tikvSnapshot) get(bo *Backoffer, k kv.Key) ([]byte, error) {
 	}
 
 	failpoint.Inject("snapshot-get-cache-fail", func(_ failpoint.Value) {
-		panic("cache miss")
+		if bo.ctx.Value("TestSnapshotCache") != nil {
```

Comment:
maybe using exp like `1*return(true)->return(false) ` like this https://github.com/pingcap/tidb/blob/2aa571b1abf329d336413fe195c88fe07c7b268c/session/session_test.go#L601 and check `failpoint.Value == true` at here is more gofail idiom~

Reply:
In the concurrent test environment, that mock panic may affect other tests.
`bo.ctx.Value("TestSnapshotCache")` is used to address that problem. @lysu

Reply:
Or we can pass a TS and panic on equal TS.

Reply:
In `failpoint.Inject` it doesn't know which TS @coocood